### PR TITLE
fix(calling): switching off stream listeners to avoid having multiple listeners

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES #https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-511584 

## This pull request addresses
While updating the local track, a DOM exception error was encountered complaining for the peer connection to be closed. Hence enabling BNR fails after the first call. This happens because the old listener is used for the OutputTrackChange event on the localAudioStream, which uses the previous media connection, which is closed while updating the local track.

index.js:6603 Uncaught (in promise) DOMException: The peer connection is closed.
    at https://localhost:8000/samples/calling.min.js:2:3712596
    at Array.forEach (<anonymous>)
    at r.value (https://localhost:8000/samples/calling.min.js:2:3712231)
    at r.value (https://localhost:8000/samples/calling.min.js:2:3713170)
    at r.value (https://localhost:8000/samples/calling.min.js:2:3847575)
    at Je.<anonymous> (https://localhost:8000/samples/calling.min.js:2:3976135)
    at Ne.emit (https://localhost:8000/samples/calling.min.js:2:3059270)
    at Xe.emit (https://localhost:8000/samples/calling.min.js:2:3061589)
    at r.changeOutputTrack (https://localhost:8000/samples/calling.min.js:2:3064449)
    at NoiseReductionEffect.t (https://localhost:8000/samples/calling.min.js:2:3065066)

Steps to reproduce:

1. Initiate a call, enable BNR then disable BNR followed by call disconnect. Everything works fine till this point.
2. Initiated 2nd call and enabled BNR, updating local track fails and it complains of peer connection being closed.

## by making the following changes
Switching off the localAudioStream listener for the OutputTrackChange event fixes this issue.

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

1. Initiate a call, enable BNR then disable BNR followed by call disconnect. Everything works fine till this point.
2. Initiated 2nd call and enabled BNR, enabling BNR works smoothly and no error is observed.

[DOM_Exception_Fix.log](https://github.com/user-attachments/files/15895528/DOM_Exception_Fix.log)

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by Shreyas281299 — APPROVED** [review]
Looks good! LGTM

File: packages/calling/src/CallingClient/calling/call.ts [hunk 1]
```diff
@@ -2488,13 +2492,12 @@ export class Call extends Eventing<CallEventTypes> implements ICall {
...
       this.localAudioStream.off(LocalStreamEventNames.EffectAdded, this.registerEffectListener);
+      this.localAudioStream.off(LocalStreamEventNames.OutputTrackChange, this.updateTrack);
     }
   }
 
   private registerListeners(localAudioStream: LocalMicrophoneStream) {
-    localAudioStream.on(LocalStreamEventNames.OutputTrackChange, (audioTrack: MediaStreamTrack) => {
-      this.mediaConnection.updateLocalTracks({audio: audioTrack});
-    });
+    localAudioStream.on(LocalStreamEventNames.OutputTrackChange, this.updateTrack);
```
Comment (sreenara) [inline packages/calling/src/CallingClient/calling/call.ts:2500]: We haven't added an expect for this statement. Is there a reason for that?

**Review by sreenara — CHANGES_REQUESTED** [review]
In the PR description where you are talking about the problem, can you please mention the sequence of events that leads to this issue?
Also, please mention the tests performed in the PR description.

File: packages/calling/src/CallingClient/calling/call.ts (hunk 1 above)
Comment (Kesari3008) [reply packages/calling/src/CallingClient/calling/call.ts:2500]: I have added it

**Comment by Kesari3008** [comment]
> In the PR description where you are talking about the problem, can you please mention the sequence of events that leads to this issue? Also, please mention the tests performed in the PR description.

PR description updated

File: packages/calling/src/CallingClient/calling/call.ts (hunk 1 above)
Comment (sreenara) [reply packages/calling/src/CallingClient/calling/call.ts:2500]: Could you point to the line where this change has been made? I am not able to find it.

**Review by sreenara — CHANGES_REQUESTED** [review]
The description of the problem still needs a change. The details that are provided in the test should be present in the problem description.
The test scenarios should contain only the observations with the fix.

**Review by sreenara — APPROVED** [review]
LGTM. Thanks for helping test this.

**Pull request merged**